http_client.go

```diff
@@ -34,35 +37,66 @@ func userFriendlyErr(errStr string) string {
 	return "HttpErr"
 }
 
-func (h *HttpClient) createPayloadRequest(uri string, jsonBytes []byte) (*http.Request, error) {
-	const JsonType = "application/json"
-	const CompressThreshold = 512
-	compressed := len(jsonBytes) > CompressThreshold
-	var payloadBuffer *bytes.Buffer
+var payloadPool = &sync.Pool{
+	New: func() interface{} {
+		return bytes.NewBuffer(nil)
+	},
+}
+
+var gzipWriterPool = &sync.Pool{
+	New: func() interface{} {
+		return gzip.NewWriter(nil)
+	},
+}
+
+const jsonType = "application/json"
+const compressThreshold = 512
+
+func (h *HttpClient) createPayloadRequest(uri string, jsonBytes []byte) (*http.Request, func(), error) {
+	compressed := len(jsonBytes) > compressThreshold
+	payloadBuffer := payloadPool.Get().(*bytes.Buffer)
+	var g *gzip.Writer
 	if compressed {
-		payloadBuffer = &bytes.Buffer{}
-		g := gzip.NewWriter(payloadBuffer)
+		g = gzipWriterPool.Get().(*gzip.Writer)
+		g.Reset(payloadBuffer)
+		defer func() {
+			if err := g.Close(); err != nil {
+				log.Printf("closing gzip writer: %v", err)
```

Comment:
Logging should be done with the logger configured on the registry. These should probably be at debug level.